src/configuration/apiConfiguration.ts

```diff
@@ -0,0 +1,183 @@
+export interface PublisherDetails {
+    eventCode: number;
+    routingKey: string;
+}
+
+export enum SubscriberEventType {
+    Update,
+    Error
+}
+
+export interface ApiConfiguration {
+    getComponentCode(componentName: string): number;
+    containsComponent(componentName: string): boolean;
+    getStateMachineCode(componentName: string, stateMachineName: string): number;
+    containsStateMachine(componentName: string, stateMachineName: string): boolean;
+    getPublisherDetails(componentCode: number, stateMachineCode: number, messageType: string): PublisherDetails;
+    getSubscriberTopic(componentCode: number, stateMachineCode: number, type: SubscriberEventType): string;
+    getSnapshotTopic(componentCode: number): string;
+    getStateName(componentCode: number, stateMachineCode: number, stateCode: number): string;
+    containsPublisher(componentCode: number, stateMachineCode: number, messageType: string): boolean;
+    containsSubscriber(componentCode: number, stateMachineCode: number, type: SubscriberEventType): boolean;
+}
+
+export class DefaultApiConfiguration implements ApiConfiguration {
+
+    private _config: any;
+
+    constructor(configObject: any) {
```

Comment:
optional: could we describe the type of this object? at least partially (i.e. the attributes we use)